java/maven.indexer/src/org/netbeans/modules/maven/indexer/CustomArtifactContextProducer.java

```diff
@@ -36,6 +37,16 @@ public CustomArtifactContextProducer(ArtifactPackagingMapper mapper) {
         super(mapper);
     }
 
+    @Override
+    protected Gav getGavFromPath(IndexingContext context, String repositoryPath, String artifactPath) {
+        // filter out maven-metadata[-repo_name].xml to avoid IndexOutOfBoundsExceptions in
+        // org.apache.maven.index.artifact.M2GavCalculator.getSnapshotGav(M2GavCalculator.java:187)
+        if (artifactPath.endsWith(".xml") && artifactPath.substring(artifactPath.lastIndexOf("/")+1).startsWith("maven-metadata")) {
+            return null;
+        }
+        return super.getGavFromPath(context, repositoryPath, artifactPath);
+    }
```

Comment:
added a workaround for the IOOBE in maven-indexer.

issue filed under:
https://issues.apache.org/jira/browse/MINDEXER-144